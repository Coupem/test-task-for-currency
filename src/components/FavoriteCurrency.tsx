// eslint-disable-next-line @typescript-eslint/no-use-before-define
import React from 'react';
import PropTypes from 'prop-types';

import { ListGroup } from 'react-bootstrap';
import {
  FavoriteCurrencyWrapper,
  NoFavoriteCurrency,
  TitleFavoriteCurrency,
  ListGroupContainer,
} from '../styled';
import { FavoriteCurrencyProp } from '../types';

const FavoriteCurrency = ({
  favoriteCurrencies,
}: FavoriteCurrencyProp): JSX.Element => (
  <FavoriteCurrencyWrapper>
    <TitleFavoriteCurrency>Favorite Currency:</TitleFavoriteCurrency>

    {Object.values(favoriteCurrencies).length === 0 && (
      <NoFavoriteCurrency>NO CURRENCY</NoFavoriteCurrency>
    )}

    {Object.values(favoriteCurrencies).map((breakpoint) => (
      <ListGroupContainer key={breakpoint.ccy}>
        <ListGroup.Item>Currency: {breakpoint.ccy}</ListGroup.Item>
        <ListGroup.Item>Buy: {breakpoint.buy}</ListGroup.Item>
        <ListGroup.Item>Sale: {breakpoint.sale}</ListGroup.Item>
      </ListGroupContainer>
    ))}
  </FavoriteCurrencyWrapper>
);

FavoriteCurrency.propTypes = {
  favoriteCurrencies: PropTypes.shape({
    ccy: PropTypes.string,
    buy: PropTypes.string,
    sale: PropTypes.string,
  }),
};

export default FavoriteCurrency;
