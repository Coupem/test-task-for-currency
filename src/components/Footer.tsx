import React from 'react';
import { FooterComponent } from '../styled';

const Footer = (): JSX.Element => (
  <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>
);

Footer.propTypes = {};

export default Footer;
