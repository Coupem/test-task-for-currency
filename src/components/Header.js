import React from 'react';
import { HeaderComponent } from '../styled';

const Header = () => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
