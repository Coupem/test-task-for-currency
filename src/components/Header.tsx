import React from 'react';
import { HeaderComponent } from '../styled';

const Header = (): JSX.Element => (
  <HeaderComponent>Best converter & valid currency course</HeaderComponent>
);

Header.propTypes = {};

export default Header;
