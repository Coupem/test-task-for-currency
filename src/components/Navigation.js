import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import { CONVERTER_URL, COURSE_URL } from '../constants';
import { NavButton, NavigationComponent } from '../styled';

const Navigation = () => {
  const history = useHistory();
  const { pathname } = useLocation();

  const onClickButtonConverter = () => {
    history.push('/');
  };

  const onClickButtonCourse = () => {
    history.push('/course');
  };

  return (
    <NavigationComponent>
      <NavButton
        primary={pathname === CONVERTER_URL}
        onClick={onClickButtonConverter}
      >
        Converter
      </NavButton>
      <NavButton
        primary={pathname === COURSE_URL}
        onClick={onClickButtonCourse}
      >
        Course
      </NavButton>
    </NavigationComponent>
  );
};

Navigation.propTypes = {};

export default Navigation;
