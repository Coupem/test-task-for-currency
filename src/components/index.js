export { default as InputValuesGroup } from './Converter/InputValuesGroup';
export { default as ConvertCurrency } from './Converter/ConvertCurrency';
export { default as FavoriteCurrency } from './Course/FavoriteCurrency';
export { default as Navigation } from './Navigation/Navigation';
export { default as Header } from './Header/Header';
export { default as Footer } from './Footer/Footer';
