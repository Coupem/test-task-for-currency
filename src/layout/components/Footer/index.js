import React from 'react';
import { FooterComponent } from './styledComponents';

function Footer() {
  return <FooterComponent>Â© 2021 My. Built My Theme.</FooterComponent>;
}

Footer.propTypes = {};

export default Footer;
