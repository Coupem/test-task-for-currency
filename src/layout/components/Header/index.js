import React from 'react';
import { HeaderComponent } from './styledComponents';

function Header() {
  return (
    <HeaderComponent>Best converter & valid currency course</HeaderComponent>
  );
}

Header.propTypes = {};

export default Header;
