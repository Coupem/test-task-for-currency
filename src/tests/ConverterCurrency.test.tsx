import { CommonWrapper, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import configureStore from 'redux-mock-store';

import { Provider } from 'react-redux';
import { ConvertCurrency } from '../components/index';

describe('Converter page', () => {
  const mockStore = configureStore();
  let wrapper:
    | CommonWrapper<{}, {}, React.Component<{}, {}, any>>
    | cheerio.Cheerio;
  let state;
  beforeAll(() => {
    state = {
      currency: {
        currency: [
          { baseCcy: 'UAH', buy: '1', ccy: 'UAH', sale: '01' },
          { baseCcy: 'USD', buy: '2', ccy: 'BTC', sale: '02' },
          { baseCcy: 'UAH', buy: '3', ccy: 'USD', sale: '03' },
          { baseCcy: 'UAH', buy: '4', ccy: 'RUR', sale: '04' },
          { baseCcy: 'UAH', buy: '5', ccy: 'EUR', sale: '05' },
        ],
        isLoadingCurrency: false,
        error: 'string',
      },
    };
    wrapper = mount(
      <Provider store={mockStore(state)}>
        <ConvertCurrency convertedValue={0} baseCurrency="" />
      </Provider>
    );
  });

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
