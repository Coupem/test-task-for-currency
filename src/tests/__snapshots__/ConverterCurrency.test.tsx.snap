// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter page should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConvertCurrency
    baseCurrency=""
    convertedValue={0}
  >
    <Styled(Component)
      bordered={true}
      hover={true}
      striped={true}
    >
      <ForwardRef
        bordered={true}
        className="sc-iemWCZ cunngp"
        hover={true}
        striped={true}
      >
        <table
          className="sc-iemWCZ cunngp table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Converted Value
              </th>
              <th>
                Currency
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="UAH"
            >
              <td
                className="p-lg-3"
              >
                0.000
              </td>
              <td
                className="w-25 p-lg-2"
              >
                UAH
              </td>
            </tr>
            <tr
              key="BTC"
            >
              <td
                className="p-lg-3"
              >
                0.000
              </td>
              <td
                className="w-25 p-lg-2"
              >
                BTC
              </td>
            </tr>
            <tr
              key="USD"
            >
              <td
                className="p-lg-3"
              >
                0.000
              </td>
              <td
                className="w-25 p-lg-2"
              >
                USD
              </td>
            </tr>
            <tr
              key="RUR"
            >
              <td
                className="p-lg-3"
              >
                0.000
              </td>
              <td
                className="w-25 p-lg-2"
              >
                RUR
              </td>
            </tr>
            <tr
              key="EUR"
            >
              <td
                className="p-lg-3"
              >
                0.000
              </td>
              <td
                className="w-25 p-lg-2"
              >
                EUR
              </td>
            </tr>
          </tbody>
        </table>
      </ForwardRef>
    </Styled(Component)>
  </ConvertCurrency>
</Provider>
`;
